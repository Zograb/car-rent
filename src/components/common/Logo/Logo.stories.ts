import type { Meta, StoryObj } from '@storybook/react';

import { Logo } from './Logo';

const meta: Meta<typeof Logo> = {
  title: 'Logo',
  component: Logo,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Logo>;

export const Basic: Story = {
  args: {
    hideLogoName: false,
  },
  parameters: {
    backgrounds: { default: 'dark' },
  },
};

export const WithoutLogoName: Story = {
  args: {
    hideLogoName: true,
  },
  parameters: {
    backgrounds: { default: 'dark' },
  },
};
