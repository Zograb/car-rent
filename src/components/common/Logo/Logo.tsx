import type { FC } from 'react';
import Image, { type ImageProps } from 'next/image';
import 'twin.macro';

// Asssets
import LogoSrc from '@/assets/images/logo.svg';

export interface LogoProps extends Omit<ImageProps, 'src' | 'alt'> {
  // hide/show logo name
  hideLogoName?: boolean;
}

const Logo: FC<LogoProps> = ({ hideLogoName = false, ...otherProps }) => (
  <div tw="flex justify-start items-center" data-testid="logo">
    <Image
      tw="h-logo-vertical"
      width={40}
      height={30}
      src={LogoSrc}
      alt={'Logo'}
      {...otherProps}
    />
    {!hideLogoName && (
      <div
        tw="ml-1 text-white text-logo font-bold"
        data-testid="logo-name"
      >
        CAR RENT
      </div>
    )}
  </div>
);

export default Logo;
