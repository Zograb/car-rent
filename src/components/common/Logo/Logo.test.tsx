import { render } from '@testing-library/react';

import Logo from './Logo';

describe('Logo', () => {
  it('should match snapshot', () => {
    const { container } = render(<Logo />);
    expect(container).toMatchSnapshot();
  });

  it('should render logo', () => {
    const { getByTestId } = render(<Logo />);
    const logo = getByTestId('logo');
    expect(logo).toBeInTheDocument();
  });

  it('should render logo without name', () => {
    const { queryByTestId } = render(<Logo hideLogoName />);
    const logoName = queryByTestId('logo-name');
    expect(logoName).not.toBeInTheDocument();
  });
});
