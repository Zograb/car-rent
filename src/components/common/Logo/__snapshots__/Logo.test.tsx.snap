// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.c1 {
  height: 1.875rem;
}

.c2 {
  margin-left: 0.25rem;
  font-size: 1.625rem;
  font-weight: 700;
  --tw-text-opacity: 1;
  color: rgb(255 255 255 / var(--tw-text-opacity));
}

<div>
  <div
    class="c0"
    data-testid="logo"
  >
    <img
      alt="Logo"
      class="c1"
      data-nimg="1"
      decoding="async"
      height="30"
      loading="lazy"
      src="logo.svg"
      style="color: transparent;"
      width="40"
    />
    <div
      class="c2"
      data-testid="logo-name"
    >
      CAR RENT
    </div>
  </div>
</div>
`;
